lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,6 +163,18 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not isinstance(encoded_text, list) or not isinstance(vocabulary, dict) or not (isinstance(
+            end_of_word_token, str) or end_of_word_token is None):
+        return None
+    decoded_tokens = []
+    for i in encoded_text:
+        for token, token_i in vocabulary.items():
+            if token_i == i and end_of_word_token is not None:
+                decoded_tokens.append(' ' if token == end_of_word_token else token)
```

Comment:
I suggest simply replacing the EoW token in the resulting string

Reply:
I'm sorry, I didn't quite understand what exactly is required of me.